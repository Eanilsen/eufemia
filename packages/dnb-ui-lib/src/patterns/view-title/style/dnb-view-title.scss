/*
* DNB ViewTitle
*
* ======================================================================== */

@import '../../../style/component/core-imports.scss';

.dnb-view-title {
  @include componentReset();
}

@import './_view-title.scss';
