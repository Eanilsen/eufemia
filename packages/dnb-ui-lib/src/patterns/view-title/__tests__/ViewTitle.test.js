/**
 * Component Test
 *
 */

import React from 'react'
import {
  shallow,
  fakeProps,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../ViewTitle'
// just to make sure we re-run the test in watch mode due to changes in this file
import '../style/dnb-view-title.scss'

const props = fakeProps(require.resolve('../ViewTitle'), {
  optional: true
})

describe('ViewTitle component', () => {
  const ComponentWrap = shallow(<Component {...props} />)
  it('have to match snapshot', () => {
    expect(toJson(ComponentWrap)).toMatchSnapshot()
  })
})

describe('ViewTitle scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-view-title.scss'))
    expect(scss).toMatchSnapshot()
  })
})
