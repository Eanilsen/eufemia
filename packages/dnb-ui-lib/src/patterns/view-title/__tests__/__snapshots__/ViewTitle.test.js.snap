// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTitle component have to match snapshot 1`] = `
<tag
  className="dnb-view-title className class"
>
  text
</tag>
`;

exports[`ViewTitle scss have to match snapshot 1`] = `
"/*
* DNB ViewTitle
*
* ======================================================================== */
/*
 * Utilities
 * ========================================================================== */
/*
 * Typography
 * ========================================================================== */
:root {
  --font-family-std: 'Fedra Sans Std', sans-serif;
  --font-family-demi: 'Fedra Sans Std Demi', sans-serif; }

.typo-std,
.typo-book {
  font-family: var(--font-family-std);
  font-weight: normal;
  font-style: normal; }

.typo-medium {
  font-family: var(--font-family-std);
  font-weight: 600;
  font-style: normal; }

.typo-demi {
  font-family: var(--font-family-demi);
  font-weight: normal;
  font-style: normal; }

.typo-number--old-style {
  font-variant-numeric: oldstyle-nums;
  -moz-font-feature-settings: \\"onum\\";
  -webkit-font-feature-settings: \\"onum\\";
  font-feature-settings: \\"onum\\"; }

.typo-number--lining {
  font-variant-numeric: lining-nums;
  -moz-font-feature-settings: \\"lnum\\";
  -webkit-font-feature-settings: \\"lnum\\";
  font-feature-settings: \\"lnum\\"; }

.typo-number--tabular {
  font-variant-numeric: tabular-nums;
  -moz-font-feature-settings: \\"tnum\\";
  -webkit-font-feature-settings: \\"tnum\\";
  font-feature-settings: \\"tnum\\"; }

.typo-number--proportional {
  font-variant-numeric: proportional-nums;
  -moz-font-feature-settings: \\"pnum\\";
  -webkit-font-feature-settings: \\"pnum\\";
  font-feature-settings: \\"pnum\\"; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Book.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Book.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Medium.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Medium.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.ttf\\") format(\\"truetype\\");
  font-weight: 600;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std Demi';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Demi.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Demi.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal; }

/*
* Scopes
* ======================================================================== */
.dnb-view-title {
  font-family: var(--font-family-std, sans-serif);
  font-size: 1rem;
  font-weight: normal;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333); }
  .dnb-view-title *,
  .dnb-view-title ::before,
  .dnb-view-title ::after {
    background-repeat: no-repeat;
    box-sizing: border-box; }
  .dnb-view-title ::before,
  .dnb-view-title ::after {
    text-decoration: inherit;
    vertical-align: inherit; }

/*
 * ViewTitle element
 *
 * ========================================================================== */
.dnb-view-title {
  margin: 1.5rem 0 1rem 0;
  font-size: 2rem;
  color: var(--color-sea-green); }
  @media screen and (min-width: 700px) {
    .dnb-view-title {
      margin: 3.5rem 0 3rem 0; } }
"
`;
